packages/next/server/dev/next-dev-server.ts

```diff
@@ -449,12 +450,13 @@ export default class DevServer extends Server {
     }
     try {
       return await super.run(req, res, parsedUrl)
-    } catch (err) {
+    } catch (error) {
       res.statusCode = 500
+      const err = isError(error) ? error : error ? new Error(error + '') : null
       try {
         this.logErrorWithOriginalStack(err).catch(() => {})
         return await this.renderError(err, req, res, pathname!, {
-          __NEXT_PAGE: err?.page || pathname,
+          __NEXT_PAGE: (isError(err) && err?.page) || pathname || '',
```

Comment:
```suggestion
          __NEXT_PAGE: err?.page || pathname || '',
```

I think this can be simplified since at this point it should be `Error | null`

Reply:
Actually now that I look again, maybe remove line 455 and change this line to 

```
(isError(err) && err?.page) || pathname || ''
```

Reply:
Didn't work because `Error` has no `page` property. The `isError` check makes it a `NextError` with `page` property